asllib/Typing.ml

```diff
@@ -396,36 +395,41 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
 
   (* Begin DisjointSlicesToPositions *)
   let disjoint_slices_to_positions loc env slices =
+    let module DI = Diet.Int in
+    let exception NonStatic in
     let eval env e =
-      match reduce_constants env e with
-      | L_Int z -> Z.to_int z
-      | _ -> fatal_from e Error.(UnsupportedExpr (Static, e))
+      match reduce_to_z_opt env e with
+      | Some z -> Z.to_int z
+      | None -> raise NonStatic
+    in
+    let interval_of_slice env slice =
+      let make_interval x y =
+        if x > y then
+          fatal_from loc @@ Error.(OverlappingSlices ([ slice ], Static))
```

Comment:
I think the type of error here is misleading. The interval is reversed, there is not issue of overlapping slices. How about a new Error.ReversedSlice?